clang/include/clang/AST/Expr.h

```diff
@@ -3416,6 +3416,8 @@ class CStyleCastExpr final
       private llvm::TrailingObjects<CStyleCastExpr, CXXBaseSpecifier *> {
   SourceLocation LPLoc; // the location of the left paren
   SourceLocation RPLoc; // the location of the right paren
+  // To indicate __cheri_foo casts in dependent contexts
+  CastKind DependentCastKind = CK_Dependent;
```

Comment:
This is probably missing some (de-)serialization code, but I don't want to add that until the I've merged the rewrite of that code that was committed upstream recently.